### PR TITLE
Make the plugin also be able to apply to initialization scripts, which enables the usage of CBA repo in settings.gradle and init.gradle

### PR DESCRIPTION
**Pull request by molian6**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,3 +79,17 @@ publishing {
 Where
 * **PROJECT_ID** is the ID of the project.
 * **REPOSITORY_ID** is the ID of the repository.
+
+### Alternatives
+
+If you need to use BuildArtifacts repositories inside your `init.gradle` or `settings.gradle`, please instead use `com.google.cloud.buildartifacts.gradle-init-plugin` by adding the following configuration to the `~/.gradle/init.gradle` file.
+
+```gradle
```
Comment (molian6) [inline README.md]: This gradle script is TBD. I cannot experiment it before I publish the gradle plugin to the portal.

File: buildartifacts-gradle-plugin/src/main/java/com/google/cloud/buildartifacts/gradle/plugin/BuildArtifactsGradlePlugin.java [hunk 2]
```diff
@@ -73,24 +73,55 @@ public void setPassword(String password) {
...
+      applyProject((Project) o);
+    } else if (o instanceof Gradle) {
+      applyGradle((Gradle) o);
+    } else if (o instanceof Settings) {
+      applySettings((Settings) o);
+    }
+  }
+
+  // The plugin for Gradle will apply CBA repo settings inside settings.gradle and build.gradle.
+  void applyGradle(Gradle gradle) {
```
Comment (Gridasov) [inline buildartifacts-gradle-plugin/src/main/java/com/google/cloud/buildartifacts/gradle/plugin/BuildArtifactsGradlePlugin.java]: I would suggest to make the internally used methods private.

Comment (molian6) [reply buildartifacts-gradle-plugin/src/main/java/com/google/cloud/buildartifacts/gradle/plugin/BuildArtifactsGradlePlugin.java]: Done

File: buildartifacts-gradle-plugin/src/main/java/com/google/cloud/buildartifacts/gradle/plugin/BuildArtifactsGradlePlugin.java [hunk 3]
```diff
@@ -73,24 +73,55 @@ public void setPassword(String password) {
...
+  }
+
+  private void modifySettings(Settings s) {
+    final PluginManagementSpec pluginManagement = s.getPluginManagement();
+    if (pluginManagement != null) {
+      pluginManagement.getRepositories().all(this::configureBuildArtifactsRepositories);
+    }
+  }
+
+  private void configureBuildArtifactsRepositories(ArtifactRepository repo)
```
Comment (kevinmdavis) [inline buildartifacts-gradle-plugin/src/main/java/com/google/cloud/buildartifacts/gradle/plugin/BuildArtifactsGradlePlugin.java]: I think this should be singular.

**Pull request merged**